app/models/classification.rb

```diff
@@ -11,12 +11,10 @@ class Classification < ApplicationRecord
 
   validates :description, :presence => true, :length => {:maximum => 255}
   validates :description, :uniqueness => {:scope => [:parent_id]}, :if => proc { |c|
-    cond = ["parent_id = ? AND description = ?", c.parent_id, c.description]
-    unless c.new_record?
-      cond.first << " AND id != ?"
-      cond << c.id
-    end
-    c.class.in_region(region_id).exists?(cond)
+    cond = c.class.in_region(region_id).where(:description => c.description)
+    cond = c.category? ? cond.is_category : cond.where(:parent_id => c.parent_id)
+    cond = cond.where.not(:id => c.id) unless c.new_record?
+    cond.exists?
```

Comment:
if this is too much, I can probably drop this.
I just wanted to get in the `is_category` scope
But if we make the transition from `0` to `nil` then this will work as is.

(string munging SQL is just so ugly)

Reply:
What does `c.category?` do?  Does it execute to query to determine that or is it funcationally similar to `is_category`?  That was the only piece in this that I didn't understand, but the rest looks good.

Reply:
```ruby
def category?
  parent_id == 0
end
```

And yes, it is not necessary

Reply:
I'll drop this

Reply:
Thinking aloud: the one non-obviously equivalent part is:
`c.category? ? cond.is_category : cond.where(:parent_id => c.parent_id)` 
"if parent_id is 0/nil, constrain to 0/nil, else to whatever parent_id we have"
OK, this works :+1:

Could you just do `.where(:parent_id => c.parent_id)`?  
Stylistically I think that'd be clean enough in this context of uniqueness (which is essentially equality).
But practically, this is necessary if you prefer temporarily accepting both 0 and nil as #18301 does :+1: